scala-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/scala/ScalaInterpreter.scala

```diff
@@ -228,16 +228,15 @@ class ScalaInterpreter() extends Interpreter {
     urls.foldLeft("")((l, r) => ClassPath.join(l, r.toString))
   }
 
-  protected def interpreterArgs(kernel: KernelLike): List[String] = {
+  protected def interpreterArgs(): List[String] = {
     import scala.collection.JavaConverters._
-    kernel.config.getStringList("interpreter_args").asScala.toList
-  }
-
-  protected def maxInterpreterThreads(kernel: KernelLike): Int = {
-    kernel.config.getInt("max_interpreter_threads")
+    if(config.hasPath("interpreter_args"))
+      config.getStringList("interpreter_args").asScala.toList
```

Comment:
https://github.com/apache/incubator-toree/blob/master/kernel/src/main/scala/org/apache/toree/boot/CommandLineOptions.scala#L145

and

https://github.com/apache/incubator-toree/blob/master/kernel/src/main/scala/org/apache/toree/boot/CommandLineOptions.scala#L171

Maybe that is causing the array to be invalidated since not providing the commandline argument is setting the value to None?